onchain/cairo/src/defi/vault.cairo

```diff
@@ -76,22 +76,56 @@ mod Vault {
         // Used the specify ratio. Burn the token. Check the pooling withdraw
         fn mint_by_token(ref self: ContractState, token_address: ContractAddress, amount: u256) {
             let caller = get_caller_address();
-        // Check if token valid
-
-        // Sent token to deposit
-
-        // let token_deposited= IERC20MintableDispatcher{ token_address};
-        // token_deposited.transfer_from(caller, get_contract_address, amount);
-
-        // Mint token and send it to the receiver
-
-        // let token_mintable= IERC20MintableDispatcher{ token_address};
-
-        // Calculate the ratio if 1:1, less or more
-        // let amount_ratio=1;
-        // // let ratio =;
-        // token_mintable.mint(caller, amount_ratio);
 
+            // Check if token valid
+            assert(self.is_token_permitted(token_address), 'Non permited token');
+
+            // Sent token to deposit
+            let token_deposited = IERC20Dispatcher { contract_address: token_address };
+            token_deposited.approve(caller, amount);
+            token_deposited.transfer_from(caller, get_contract_address(), amount);
+
+            // Mint token and send it to the receiver
+            let token_mintable = IERC20MintableDispatcher {
+                contract_address: self.token_address.read()
+            };
+            let token = self.token_permitted.read(token_address);
+
+            // Calculate the ratio if 1:1, less or more
+            let amount_ratio = token.ratio_mint * amount;
```

Comment:
Please comment with a TODO, and just add the amount without the ratio at the moment